### PR TITLE
Added include_metadata feature and some fixes

### PR DESCRIPTION
**Pull request by muhammad-ali-e**
- Fixes and Solutions for the issues:

  - https://github.com/Zipstack/unstract-python-client/issues/4
  - https://github.com/Zipstack/unstract-python-client/issues/5
  - https://github.com/Zipstack/unstract-python-client/issues/6

- Updated pre-commit by changing max line to 88

### CODE REVIEW

**Review by jaseemjaskp — APPROVED** [review]
LGTM

File: README.md [hunk 1]
```diff
@@ -58,6 +59,12 @@ except APIDeploymentsClientException as e:
     print(e)
 ```
 
+## Parameter Details
+
+`api_timeout`: Set a timeout for API requests, e.g., `api_timeout=10`.
+`logging_level`: Set logging verbosity (e.g., "`DEBUG`").
+`include_metadata`: If set to `True`, the response will include additional metadata for each calls in the structured output, providing context of the results.
```
Comment (chandrasekharan-zipstack) [inline README.md:66]:
```suggestion
`include_metadata`: If set to `True`, the response will include additional metadata (cost, tokens consumed and context) for each call made by the Prompt Studio exported tool.
```
The user might not know what `structured output`

File: src/unstract/api_deployments/client.py [hunk 2]
```diff
@@ -141,47 +139,71 @@ def structure_file(self, file_paths: list[str]) -> dict:
...
+        # The returned object is wrapped in a "message" key.
+        # Let's simplify the response.
         obj_to_return = {}
 
-        if response.status_code == 401:
+        try:
+            response_data = response.json()
+            response_message = response_data.get("message", {})
+        except JSONDecodeError:
+            self.logger.error("Failed to decode JSON response", exc_info=True)
```
Comment (ritwik-g) [inline src/unstract/api_deployments/client.py:150]: @muhammad-ali-e should we include `response` or `response.text` in case of error for debugging?

**Review by ritwik-g — APPROVED** [review]
@muhammad-ali-e @hari-kuriakose @jaseemjaskp  changes looks good. But this will technically be a breaking change compared to our previous return object. 

So let's bump the major version when we publish this

File: src/unstract/api_deployments/client.py (hunk 2 above)
Comment (muhammad-ali-e) [reply src/unstract/api_deployments/client.py]: Updated

**Pull request merged**